pkg/networkplugin-syncer/main.go

```diff
@@ -51,6 +55,29 @@ func main() {
 	klog.Info("All controllers stopped or exited. Stopping submariner-networkplugin-syncer")
 }
 
+func getK8sClient() kubernetes.Interface {
+	var cfg *rest.Config
+	var err error
+	if masterURL == "" && kubeconfig == "" {
+		cfg, err = rest.InClusterConfig()
+		if err != nil {
+			klog.Fatalf("Error getting in-cluster-config, please set kubeconfig && master parameters")
+		}
+	} else {
```

Comment:
Doesn't `clientcmd.BuildConfigFromFlags` do this?

Reply:
Yes but it throws out a big warning when we start every daemon. Probably a common "getRestConfig" for submariner could be useful in avoiding such warnings everywhere.

Look:
```
unc BuildConfigFromFlags(masterUrl, kubeconfigPath string) (*restclient.Config, error) {
	if kubeconfigPath == "" && masterUrl == "" {
		klog.Warningf("Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.")
		kubeconfig, err := restclient.InClusterConfig()
		if err == nil {
			return kubeconfig, nil
		}
		klog.Warning("error creating inClusterConfig, falling back to default config: ", err)
	}
	return NewNonInteractiveDeferredLoadingClientConfig(
		&ClientConfigLoadingRules{ExplicitPath: kubeconfigPath},
		&ConfigOverrides{ClusterInfo: clientcmdapi.Cluster{Server: masterUrl}}).ClientConfig()
}
```

Reply:
ahh ok.